File: java/random/src/main/java/random/RandomNumberGame.java
package random;

import java.util.Random;
import java.util.Scanner;

public class RandomNumberGame {
  public static void main(String[] args) {
    Random rand = new Random();
    boolean win = false;
    int numberToGuess = rand.nextInt(1000) + 1;
    int numberOfTries = 0;

    System.out.println("============== RANDOM NUMBER GAME ==============");
    System.out.println("Welcome to the Random Number Game!");
    System.out.println("Enter 'exit' to quit the game.");

    try (Scanner input = new Scanner(System.in)) {
      int guess = 0;

      while (!win) {
        try {
          numberOfTries++;
          System.out.print("Guess a number between 1 and 1000 : ");
          String line = input.nextLine();
          if (line.trim().isEmpty()) {
            System.out.println("You must enter a number.");
            continue;
          } else if (line.equalsIgnoreCase("exit")) {
            if (numberOfTries == 1) {
              System.out.println("You didn't even try!");
            } else {
              System.out.println("You gave up after " + numberOfTries + " tries.");
            }
            System.out.println("Goodbye!");
            break;
          }

          guess = Integer.parseInt(line);

          if (guess == numberToGuess) {
            win = true;
          } else if (guess < numberToGuess) {
            System.out.println("Too low! Try again.");
          } else if (guess > numberToGuess) {
            System.out.println("Too high! Try again.");
          }
        } catch (NumberFormatException e) {
          System.out.println("Invalid input. Please enter a number.");
        }
      }
    }

    if (numberOfTries == 1 && win) {
      System.out.println("Amazing! You guessed the number on first try!");
    } else if (win) {
      System.out.println("You win!");
      System.out.println("The number was " + numberToGuess);
      System.out.println("It took you " + numberOfTries + (numberOfTries > 1 ? " tries." : " try."));
    }

    System.out.println("================================================");
  }
}

File: c++/random/src/randomNumberGame.cpp
#include <cstdlib>
#include <ctime>
#include <iostream>
#include <string>

int main() {
  srand(time(0));
  bool win = false;
  int numberToGuess = rand() % 1000 + 1;
  int numberOfTries = 0;

  std::cout << "============== RANDOM NUMBER GAME ==============\n";
  std::cout << "Welcome to the Random Number Game!\n";
  std::cout << "Enter 'exit' to quit the game.\n";

  std::string line;
  int guess = 0;

  while (!win) {
    numberOfTries++;
    std::cout << "Guess a number between 1 and 1000 : ";
    std::getline(std::cin, line);

    if (line.empty()) {
      std::cout << "You must enter a number.\n";
      continue;
    } else if (line == "exit") {
      if (numberOfTries == 1) {
        std::cout << "You didn't even try!\n";
      } else {
        std::cout << "You gave up after " << numberOfTries << " tries.\n";
      }
      std::cout << "Goodbye!\n";
      break;
    }

    try {
      guess = std::stoi(line);
    } catch (std::invalid_argument &) {
      std::cout << "Invalid input. Please enter a number.\n";
      continue;
    }

    if (guess == numberToGuess) {
      win = true;
    } else if (guess < numberToGuess) {
      std::cout << "Too low! Try again.\n";
    } else if (guess > numberToGuess) {
      std::cout << "Too high! Try again.\n";
    }
  }

  if (numberOfTries == 1 && win) {
    std::cout << "Amazing! You guessed the number on first try!\n";
  } else if (win) {
    std::cout << "You win!\n";
    std::cout << "The number was " << numberToGuess << "\n";
    std::cout << "It took you " << numberOfTries
              << (numberOfTries > 1 ? " tries.\n" : " try.\n");
  }

  std::cout << "================================================\n";

  return 0;
}
